src/ssh/azext_ssh/resource_tag_utils.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core import azclierror
+from knack import log
+
+logger = log.get_logger(__name__)
+# Checking Port and Resource Tag argument and fetching appropiate tag
+
+
+def _handle_os_type_for_tag(cmd, op_info):
+    tags = {}
+    server_error = False
+
+    # Warning about two parameters for port (Port, Resource Tag)
+    if op_info.port and op_info.resource_tag:
+        logger.warning("Warning: Both --port and --resource-tag arguments were specified."
+                            "The --port option will take precedence and the --resource-tag will be ignored."
+                            "To use the port number from the --resource-tag, please omit the --port argument.")
+        return
+    # Only port is specified, use port
+    if op_info.port:
+        return
+    # Retrieving machine's resource tags
+    if op_info.resource_type.lower() == "microsoft.compute/virtualmachines":
+        tags, server_error = _check_azure_vm_tag(cmd, op_info.resource_group_name, op_info.vm_name, tags)
+    elif op_info.resource_type.lower() == "microsoft.hybridcompute/machines":
+        tags, server_error = _check_arc_server_tag(cmd, op_info.resource_group_name, op_info.vm_name, tags)
+    elif op_info.resource_type.lower() == "microsoft.connectedvmwarevsphere/virtualmachines":
+        tags, server_error = _check_connected_vmware_tag(cmd, op_info.resource_group_name, op_info.vm_name, tags)
+
+    if server_error:
+        logger.warning("Failed to retrieve resource tags from the server. Using default port 22.")
+        op_info.port = "22"
+        return
+    # Checking for proper Resource Tag configurations
+    def validate_and_set_port(tag_name="SSHPort"):
+        port_num = tags.get(tag_name)
+        if port_num and port_num.isdigit() and int(port_num) < 65535:
+            op_info.port = port_num
+
+        else:
+            raise azclierror.InvalidArgumentValueError(
+                f"Port '{port_num}' from resource tag '{tag_name}' was used for this command. "
+                "If this is incorrect, use the --port parameter or contact your administrator to correct the resource tag value. "
+                "Port numbers must not be empty, must not contain letters or special characters, and cannot exceed 65535.")
+    # Checking if there is a specified resource tag
+    if op_info.resource_tag:
```

Comment:
Check if resource tag is supported before making call to azure. If we don't need to retrieve resource tag, we should skip the expensive operations.

Reply:
I dont understand this comment. The call is made before because you need to get the tags to then check if the passed in resource_tag parameter is in those resource tags